libs/execution_context/ebpf_native.c

```diff
@@ -1222,7 +1228,7 @@ ebpf_native_load_programs(
     module = *existing_module;
     module_state = ebpf_lock_lock(&module->lock);
     native_lock_acquired = true;
-    if (module->loading || module->loaded) {
+    if (module->state == MODULE_STATE_LOADING || module->state == MODULE_STATE_LOADED) {
```

Comment:
This should be `if (module->state != MODULE_STATE_INITIALIZED || module->detaching)`